### ISSUE TITLE
`no-unused-vars` rule doesn't work for JSX or TSX

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#712]
### Lint Name

no-unused-vars

### Code Snippet

```ts
const foo = 42;
```

### Expected Result

`no-unused-vars` rule is triggered regardless of the file type (extension).

### Actual Result

When the preceding snippet is saved as `foo.ts` or `foo.js`, the rule works as expected:

```
(no-unused-vars) `foo` is never used
const foo = 42;
      ^^^
    at /home/yusuktan/Repos/github.com/magurotuna/deno_lint/foo.ts:1:6

    hint: If this is intentional, prefix it with an underscore like `_foo`

Found 1 problem
Checked 1 file
```
But no diagnostic appears when saved as `foo.tsx` or `foo.jsx`:

```
Checked 1 file
```

### Additional Info

nothing

### Version

deno 1.10.3 (release, x86_64-unknown-linux-gnu)
v8 9.1.269.27
typescript 4.2.2

**Comment by magurotuna** [comment]
Oh, I was stupid, `no-unused-vars` skips linting jsx / tsx files _intentionally_ to avoid false positives around the factory functions...
https://github.com/denoland/deno_lint/blob/1c2ba8e10ce9f6e5406047e6165cf277aa2b92cc/src/rules/no_unused_vars.rs#L59-L64
